src/NServiceBus.Core/Transports/Learning/LearningTransportInfrastructure.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using DelayedDelivery;
+    using Performance.TimeToBeReceived;
+    using Routing;
+    using Settings;
+    using Transport;
+
+    class LearningTransportInfrastructure : TransportInfrastructure
+    {
+        public LearningTransportInfrastructure(SettingsHolder settings)
+        {
+            this.settings = settings;
+
+            if (!settings.TryGet(StorageLocationKey, out storagePath))
+            {
+                var solutionRoot = FindSolutionRoot();
+                storagePath = Path.Combine(solutionRoot, ".learningtransport");
+            }
+        }
+
+        public override IEnumerable<Type> DeliveryConstraints { get; } = new[]
+        {
+            typeof(DiscardIfNotReceivedBefore),
+            typeof(NonDurableDelivery),
```

Comment:
@timbussmann I added this to get the core tests passing. We could of course move the tests to msmq(and copy to rabbit and asb) but I called YAGNI since we want to push this down to transports anyway I figured we could live with this for now?